warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -2763,8 +2763,10 @@ public Tuple2<CloseableIterable<QueryPlan>,Boolean> getQueryRanges(ScannerFactor
             }
         }
 
-        if (config.isSortQueryBeforeGlobalIndex()) {
+        if (config.isSortQueryPreIndexWithFieldCounts()) {
             config.setQueryTree(timedSortQueryBeforeGlobalIndex(config, getMetadataHelper()));
+        } else if (config.isSortQueryPreIndexWithImpliedCounts()) {
+            config.setQueryTree(OrderByCostVisitor.order(config.getQueryTree()));
```

Comment:
Should this second ordering option be timed as well?

Reply:
Yes, yes it should. Thanks!